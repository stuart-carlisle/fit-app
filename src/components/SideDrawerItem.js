import React from 'react'
import { Link } from 'react-router-dom'

export const SideDrawerItem = ({ data, drawerToggleClickHandler }) => {
  const link = data.link
  const name = data.name
  const image = data.image
  
  const onClick = () => {
      drawerToggleClickHandler()
   }

   return(
     <Link className={"side-draw__list-item"} to={link} onClick={onClick}>
      <span className="side-draw__span-container"> 
         <img className="side-draw-image__icon" src={image} />
         <div>
            <p className="side-draw-item__title">{name}</p>
         </div>
      </span> 
     </Link>
   )

} 

export default SideDrawerItem