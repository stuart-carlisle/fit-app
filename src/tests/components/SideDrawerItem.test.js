import { shallow } from 'enzyme'
import React from 'react'
import SideDrawerItem from '../../components/SideDrawerItem'
 
test('Should render the sidedrawer back button correctly',()=>{
    const drawerToggleClickHandler = jest.fn()
    const data = {
        link:'/dashboard',
        name:'dashboard',
        image:'test.png'
    }
    const wrapper = shallow(<SideDrawerItem data={data} drawerToggleClickHandler={drawerToggleClickHandler} />)
    expect(wrapper).toMatchSnapshot()
})

test('Should call drawer toggle click handler on click',()=>{
    const drawerToggleClickHandler = jest.fn()
    const data = {
        link:'/dashboard',
        name:'dashboard',
        image:'test.png'
    }
    const wrapper = shallow(<SideDrawerItem data={data} drawerToggleClickHandler={drawerToggleClickHandler} />)
    wrapper.find('Link').simulate('click')
    expect(drawerToggleClickHandler).toHaveBeenCalled()
})