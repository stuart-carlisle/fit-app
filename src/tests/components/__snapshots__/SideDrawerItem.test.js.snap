// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the sidedrawer back button correctly 1`] = `
<Link
  className="side-draw__list-item"
  onClick={[Function]}
  to="/dashboard"
>
  <span
    className="side-draw__span-container"
  >
    <img
      className="side-draw-image__icon"
      src="test.png"
    />
    <div>
      <p
        className="side-draw-item__title"
      >
        dashboard
      </p>
    </div>
  </span>
</Link>
`;
